tools/lintapidiff.ml

```diff
@@ -125,19 +131,26 @@ module Ast = struct
       let self = add_item ~f path inherits in
       match ty.pmty_desc with
       | Pmty_signature lst -> List.fold_left self map lst
-      | Pmty_functor ({txt;_}, _, m) ->
-          let path = Path.Papply(path, Path.Pident (Ident.create txt)) in
+      | Pmty_functor ((Named ({txt = Some txt0},_)),m) ->
+          let new_name = Ident.create_persistent txt0 in
```

Comment:
`new_name` is maube questionable as a `name` since the name part of the identifier is the same as before. Maybe more `let ident`?